lib/scanner/csv-scanner/csv-scanner.c

```diff
@@ -392,43 +392,40 @@ csv_scanner_is_scan_finished(CSVScanner *self)
 }
 
 void
-csv_scanner_input(CSVScanner *self, const gchar *input)
+csv_scanner_iterator_init(CSVScannerIterator *iterator, const gchar *input)
 {
-  self->src = input;
-  self->current_column = NULL;
+  iterator->src = input;
+  iterator->current_value = g_string_sized_new(128);
+  iterator->current_column = NULL;
```

Comment:
Good point thanks. Let's do this change in a separate changeset after pr is merged.
I created https://github.com/balabit/syslog-ng/pull/1421 so not to forget.